tests/lib/rule-tester/rule-tester.js

```diff
@@ -1897,6 +1897,7 @@ describe("RuleTester", () => {
             invalid: [{ code: "foo", errors: [{ messageId: "avoidFoo" }] }]
         });
     });
+
```

Comment:
Do we need to worry about the pathological edge case where the placeholder data contains another placeholder? It would hit your new exception unless we could escape it somehow. At first, I figured this wouldn't be a realistic situation, but maybe it could be if someone was writing a lint rule that was linting placeholders themselves. However we decide to handle it, it would be good to document the expected behavior in the form of a test case.